### PR TITLE
Cosine annealing warm restarts add gamma

### LINKED ISSUE #62640: add gamma to CosineAnnealingWarmRestarts so max lr can decrease in cycles
## 🚀 Feature

add gamma to CosineAnnealingWarmRestarts so max lr can decrease in cycles

## Motivation

In current [implementation](https://pytorch.org/docs/stable/generated/torch.optim.lr_scheduler.CosineAnnealingWarmRestarts.html), 
the max lr is the same in each cycle. It would be great if the max lr (`\eta_max` )can be set to decreasing.

## Pitch

add an `gamma` option to `CosineAnnealingWarmRestarts`, so the max learning rate decreases after each cycle

## Additional context

Similar implementation: https://github.com/katsura-jp/pytorch-cosine-annealing-with-warmup#args

I'll be glad to send a PR

### PR DESCRIPTION
**Pull request by JohnNesbit**
Fixes #62640 
Also makes the implementation of the scheduler more efficient and intuitive. The scheduler has the same output without adding a gamma.

### CODE REVIEW

**Comment by VitalyFedyunin** [comment]
Hi @@iramazanli can you please review this one?

**Comment by iramazanli** [comment]
Hey, thank you for preparing this PR and working on this. Currently, we are working / discussing re-designing all learning rate schedulers, and soon it's likely that we will do fundamental changes to the implementation of them. We will definitely consider this work as well in newly designed schedulers.

In that respect, i would want to wait a bit before merging this work, to finalize design process for schedulers.
